app/resources/admin.html.template

```diff
@@ -68,38 +69,74 @@
     //find all languages that the user specified with "Use?" checkboxes and
     //their associated titles.  Only allows titles from used languages.
     var used_languages = [];
-    var languages_to_titles = {}
+
     //there is a title row in the table, so the dropdowns are 1-indexed
-    for (var dropdown_index = 1; $('language_selection_' + dropdown_index) !=
+    for (var dropdown_index = 1; $('repo_titles_language_selection_' + dropdown_index) !=
          null; dropdown_index++) {
-      var language_code = $('language_selection_' + dropdown_index).value;
-      var language_title = $('language_title_' + dropdown_index).value;
+      var language_code = $('repo_titles_language_selection_' + 
+                             dropdown_index).value;
       if (language_code === 'unused') {
         continue;
       }
-      if (arr_contains(used_languages, language_code)) {
-        alert("You entered the same language (" + language_code + ") twice. " +
-              "That's not allowed.");
-        return false;
-      }
       used_languages.push(language_code);
-      languages_to_titles[language_code] = language_title;
     }
 
+    var filtered_index = [];
+    used_languages.filter(function(element, index, self){
+      if (self.indexOf(element) === index){
+        filtered_index.push(index);
+      } else {
+        used_languages.splice(index, 1);
+      }
+    });
+
     // Copy the dropdowns and the titles into the text boxes that will
     // actually be POSTed (language_menu_options and repo_titles).
     $('language_menu_options').value = JSON.stringify(used_languages);
-    $('repo_titles').value = JSON.stringify(languages_to_titles);
+    var languages_to_values = {};
+
+    for (var id_num = 0; id_num < CUSTOM_MESSAGE_TABLE_IDS.length; id_num++){
+      var table_name = CUSTOM_MESSAGE_TABLE_IDS[id_num];
+      for (var i = 0; i < filtered_index.length; i++){
+        var index = filtered_index[i];
+        var language_value = $(table_name + '_language_value_' + (index+1)).value;
+        var used_language = used_languages[index];
+        languages_to_values[used_language] = language_value;
+      }
+      $(table_name).value = JSON.stringify(languages_to_values);
+    }
 
     return true;
   }
 
+  function check_language_selected(row_number) {
```

Comment:
validate_language_selected may be a clearer name. "Validate" means checking if the input is valid. "Check" is more ambiguous.